src/boot/efi/boot.c

```diff
@@ -2357,6 +2357,9 @@ static EFI_STATUS image_start(
         if (err != EFI_SUCCESS)
                 return log_error_status(err, "Error preparing initrd: %m");
 
+        /* Ask shim to leave its protocol around, so that the stub can use it to validate PEs. */
+        shim_retain_protocol();
```

Comment:
what is the precise effect of this? why doesn't shim leave the protocol around anyway? I am trying to grok why we need this, and why this is an opt-in thing really.

Reply:
This is a local experiment, there's no such thing upstream as mentioned in the other comment. Hence the "draft" state. I've asked the shim maintainers whether this or any other solution is workable. I do not know why the protocol is uninstalled, I've asked.

Reply:
The idea was for shim to do its thing and get out of the way, but it seems it will be acceptable to have an opt-in t keep the protocol around like this